heron_base/launch/base.launch

```diff
@@ -1,75 +1,101 @@
-  </node>
+  <include file="$(find heron_description)/launch/description.launch">
+    <arg name="namespace" value="$(arg namespace)" />
+  </include>
 
-  <!-- Fix these static transforms -->
-  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster"
-    args="0.2921 0 0 0 0 0 1 base_link navsat 100" />
+  <group ns="/$(arg namespace)">
 
-    <node pkg="tf" type="static_transform_publisher" name="base_to_imu"
-      args="0.1397 0 0 0 0 0 1 imu_link base_link 100" />
+    <arg name="port" default="/dev/arduino" />
+    <node pkg="rosserial_server" type="serial_node" name="rosserial_server">
+      <param name="port" value="$(arg port)" />
+      <remap from="has_wifi" to="wireless/connected" />
+    </node>
 
-  <!-- TODO: Eliminate this by compiling the necessary message definitions into heron_base. -->
-  <node pkg="rosserial_python" type="message_info_service.py" name="rosserial_message_info" />
+    <!-- TODO: Eliminate this by compiling the necessary message definitions into heron_base. -->
+    <node pkg="rosserial_python" type="message_info_service.py" name="rosserial_message_info" />
 
-  <group if="$(optenv USE_SNMP_CHECK_NODE true)">
-    <node pkg="heron_base" type="snmp_lights.py" name="wifi_lights">
-	<param name="~ip_addr" value="$(optenv SNMP_IP 192.168.1.9)" />
-    </node>
-  </group>
+    <group if="$(optenv USE_SNMP_CHECK_NODE true)">
+      <node pkg="heron_base" type="snmp_lights.py" name="wifi_lights">
+        <param name="~ip_addr" value="$(optenv SNMP_IP 192.168.1.9)" />
+      </node>
+    </group>
 
-  <group unless="$(optenv USE_SNMP_CHECK_NODE false)">
-    <node ns="wireless" pkg="wireless_watcher" name="wireless_watcher" type="watcher_node" respawn="true" />
-  </group>
+    <group unless="$(optenv USE_SNMP_CHECK_NODE true)">
+        <node ns="wireless" pkg="wireless_watcher" name="wireless_watcher" type="watcher_node" respawn="true" />
+    </group>
+
+    <group if="$(eval arg('namespace') == '')">
+     <node pkg="um6" type="um6_driver" name="um6_driver" output="screen">
+       <param name="tf_ned_to_enu" value="true"/>
+       <param name="/um6_driver/port" value="$(optenv HERON_IMU_UM6_PORT /dev/clearpath/imu)" />
+       <remap from="imu/data" to="imu/data_raw" />
+       <param name="lt_model" value="true" />
+       <param name="linear_acceleration_stdev" value="0.38" />
+       <param name="frame_id" value="imu_link" />
+     </node>
+    </group>
+
+    <group unless="$(eval arg('namespace') == '')">
+     <node pkg="um6" type="um6_driver" name="um6_driver" output="screen">
+       <param name="tf_ned_to_enu" value="true"/>
+       <param name="/um6_driver/port" value="$(optenv HERON_IMU_UM6_PORT /dev/clearpath/imu)" />
+       <remap from="imu/data" to="imu/data_raw" />
+       <param name="lt_model" value="true" />
+       <param name="linear_acceleration_stdev" value="0.38" />
+       <param name="frame_id" value="$(arg namespace)/imu_link" />
```

Comment:
Could an envar just be used here instead of the grouping?

Reply:
Could it just be `"$(arg namespace)/imu_link"`

Reply:
I'm not sure if that gives a double ```/``` or not but even a single ```/``` is problematic. It results in:

``` Error was Invalid argument "/imu_link" passed to lookupTransform argument source_frame in tf2 frame_ids cannot start with a '/' like..."```

Reply:
Ah yes, you are correct!